### PR TITLE
ADDED POST MODEL

### PR DESCRIPTION
**Pull request by NickW2000**
this is post model

### CODE REVIEW

**Review by Malachi304 — COMMENTED** [review]
Your schema looks good 
for comments, but put it in a separate comment file to separate concerns

File: backend/models/comment.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Comment Schema (for reference, you might have a separate schema for comments)
+const commentSchema = new mongoose.Schema({
+    userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
```
Comment (angel908fs) [inline backend/models/comment.js:3]: for consistency with user.js, let's use user_id instead

File: backend/models/comment.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    content: { type: String, required: true },
+    createdAt: { type: Date, default: Date.now }
```
Comment (angel908fs) [inline backend/models/comment.js:5]: created_at

File: backend/models/post.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+
+// Post Schema
+const postSchema = new mongoose.Schema(
+    {
+        post_id: { auto: true }, // Automatically generated ObjectId for the post
+        song_link: { type: String, required: true }, // Link to the song
+        likes: { type: Number, default: 0 }, // Count of likes
+        comments: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Comment' }], // List of comment IDs
+        userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true } // Reference to the user who created the post
```
Comment (angel908fs) [inline backend/models/post.js:12]: user_id

File: backend/models/post.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+
+
+// Post Schema
+const postSchema = new mongoose.Schema(
+    {
+        post_id: { auto: true }, // Automatically generated ObjectId for the post
+        song_link: { type: String, required: true }, // Link to the song
+        likes: { type: Number, default: 0 }, // Count of likes
+        comments: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Comment' }], // List of comment IDs
```
Comment (angel908fs) [inline backend/models/post.js:11]: you might want to change this to the type of the user_id field under the user model

**Review by angel908fs — APPROVED** [review]

**Pull request merged**